File: 0001_0100/0030/0030.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s = [int(x) for x in f.readline().strip().split(':')]
    t = [int(x) for x in f.readline().strip().split(':')]
    p = [0] * 10
    while s <= t:
        for x in s:
            p[x // 10] += 1
            p[x % 10] += 1
        s[2] += 1
        if s[2] == 60:
            s[2] = 0
            s[1] += 1
            if s[1] == 60:
                s[1] = 0
                s[0] += 1
    for i in p:
        q.write(str(i) + '\n')

File: 0001_0100/0030/0030.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var a [10]int
	var s, t [3]int
	fmt.Fscanf(f, "%d:%d:%d\n%d:%d:%d", &s[0], &s[1], &s[2], &t[0], &t[1], &t[2])
	for {
		for _, v := range s {
			a[v/10]++
			a[v%10]++
		}
		if s == t {
			break
		}
		if s[2]++; s[2] == 60 {
			s[2] = 0
			if s[1]++; s[1] == 60 {
				s[1] = 0
				s[0]++
			}
		}
	}
	for _, v := range a {
		fmt.Fprintln(q, v)
	}
}
